src/Solver/Numerov.cpp

```diff
@@ -0,0 +1,249 @@
+    }
+ 
+    // For each separated potential values...
+    for (int potential_counter = 0; potential_counter < potentials.size(); potential_counter++)
+    {
+        std::vector<double> &this_potential_values = potentials[potential_counter];
+
+        // The final solution energy is the sum of each energy
+        this->solutionEnergy += this->findEnergy(e_min, e_max, e_step, this_potential_values);
+        this->probability     = this->findProbability();
+        this->normalize();
+        this->multiplyWavefunction();
+    }
+    return this->solutionEnergy;
+}
+
+// Applies a bisection algorith to the numerov method to find
+// the energy that gives the non-trivial (non-exponential) solution
+// with the correct boundary conditions
+double Numerov::bisection(double e_min, double e_max, std::vector<double> &this_potential)
+{
+    double energy_middle, fx1, fb, fa;
+    std::cout.precision(17);
+
+    // The number of iterations that the bisection routine needs can be evaluated in advance
+    int itmax = ceil(log2(e_max - e_min) - log2(err)) - 1;
+
+    for (int i = 0; i < itmax; i++)
+    {
+        energy_middle = (e_max + e_min) / 2.0;
+
+        this->findWavefunction(energy_middle, this_potential);
+        fx1 = this->tempWavefunction[this->nbox];
+
+        this->findWavefunction(e_max, this_potential);
+        fb = this->tempWavefunction[this->nbox];
+
+        if (std::abs(fx1) < err)
+        {
+            return energy_middle;
+        }
+
+        if (fb * fx1 < 0.)
+        {
+            e_min = energy_middle;
+        }
+        else
+        {
+            this->findWavefunction(e_min, this_potential);
+            fa = this->tempWavefunction[this->nbox];
+
+            if (fa * fx1 < 0.)
+                e_max = energy_middle;
+        }
+    }
+
+    std::cerr << "ERROR: Solution not found using the bisection method, " << this->tempWavefunction[this->nbox] << " > " << err << std::endl;
+    return energy_middle;
+}
+
+void Numerov::printToFile()
+{
+    if (this->wavefunction.size() > 0) {
+        std::ofstream myfile("wavefunction.dat");
+        if (myfile.is_open())
+        {
+
+            for (int i = 0; i < this->wavefunction.size(); i++)
+            {
+                myfile << i << " " << this->wavefunction.at(i) << std::endl;
+            }
+            myfile.close();
+        }
+    }
+
+    if (this->probability.size() > 0) {
+        std::ofstream myfile("probability.dat");
+        if (myfile.is_open())
+        {
+
+            for (int i = 0; i < this->probability.size(); i++)
+            {
+                myfile << i << " " << this->probability.at(i) << std::endl;
+            }
+            myfile.close();
+        }
+    }
+
+}
+
+
+std::vector<double> Numerov::findProbability() {
+    std::vector<double> probability = this->tempWavefunction; 
+    std::transform(probability.begin(), probability.end(), probability.begin(), square);
+    return probability;
+}
+
+void Numerov::normalize(){
```

Comment:
this as well

Reply:
The probability is an object property, so the method take it using `this->`